tests/stack/test_api.py

```diff
@@ -0,0 +1,247 @@
+}
+
+
+@pytest.fixture()
+def stack_api():
+    workspace_api_mock = mock.patch('databricks_cli.stack.api.WorkspaceApi')
+    jobs_api_mock = mock.patch('databricks_cli.stack.api.JobsApi')
+    dbfs_api_mock = mock.patch('databricks_cli.stack.api.DbfsApi')
+    workspace_api_mock.return_value = mock.MagicMock()
+    jobs_api_mock.return_value = mock.MagicMock()
+    dbfs_api_mock.return_value = mock.MagicMock()
+    stack_api = api.StackApi(mock.MagicMock())
+    yield stack_api
+
+
+class TestStackApi(object):
+    def test_read_config(self, stack_api, tmpdir):
+        """
+            Test reading a stack configuration template
+        """
+        stack_path = os.path.join(tmpdir.strpath, TEST_STACK_PATH)
+        os.makedirs(os.path.dirname(stack_path))
+        with open(stack_path, "w+") as f:
+            json.dump(TEST_STACK, f)
+        config = stack_api._parse_config_file(stack_path)
+        assert config == TEST_STACK
+
+    def test_load_status(self, stack_api, tmpdir):
+        """
+            Test reading and parsing a deployed stack's status JSON file.
+        """
+        config_path = os.path.join(tmpdir.strpath, 'test.json')
+        status_path = stack_api._generate_stack_status_path(config_path)
+        with open(config_path, "w+") as f:
+            json.dump(TEST_STACK, f)
+        with open(status_path, "w+") as f:
+            json.dump(TEST_STATUS, f)
+        status = stack_api._load_stack_status(status_path=status_path)
+        assert status == TEST_STATUS
+        assert stack_api.deployed_resource_config == TEST_STATUS[api.STACK_RESOURCES]
+        assert all(resource[api.RESOURCE_ID] in stack_api.deployed_resources
+                   for resource in TEST_STATUS[api.STACK_DEPLOYED])
+
+    def test_default_status_path(self, stack_api, tmpdir):
+        config_path = os.path.join(tmpdir.strpath, 'test.json')
+        expected_status_path = os.path.join(tmpdir.strpath, 'test.deployed.json')
+        generated_path = stack_api._generate_stack_status_path(config_path)
+        assert expected_status_path == generated_path
+
+        config_path = os.path.join(tmpdir.strpath, 'test.st-ack.json')
+        expected_status_path = os.path.join(tmpdir.strpath, 'test.st-ack.deployed.json')
+        generated_path = stack_api._generate_stack_status_path(config_path)
+        assert expected_status_path == generated_path
+
+    def test_store_status(self, stack_api, tmpdir):
+        config_path = os.path.join(tmpdir.strpath, 'test.json')
+        status_path = stack_api._generate_stack_status_path(config_path)
+        test_data = {'test': 'test'}
+        stack_api._store_stack_status(status_path, test_data)
+
+        status = stack_api._load_stack_status(status_path)
+        assert status == test_data
+        assert os.path.exists(status_path)
+
+    def test_relative_paths(self, stack_api, tmpdir):
+        """
+            Test that the current working directory when deploying or downloading resource is same
```

Comment:
This tests that when a resource is being deployed, the current working directory is the same directory as where the stack config template lies, so that relative paths for resources work well.